src/test/scala/firrtlTests/options/phases/WriteOutputAnnotationsSpec.scala

```diff
@@ -100,9 +109,59 @@ class WriteOutputAnnotationsSpec extends AnyFlatSpec with Matchers with firrtl.t
     out.toSeq should be (annotations)
   }
 
+  it should "write CustomFileEmission annotations" in new Fixture {
+    val file = new File("write-CustomFileEmission-annotations.anno.json")
+    val annotations = Seq( TargetDirAnnotation(dir),
+                           OutputAnnotationFileAnnotation(file.toString),
+                           WriteOutputAnnotationsSpec.Custom("hello!") )
+    val serializedFileName = view[StageOptions](annotations).getBuildFileName("Custom", Some(".Emission"))
+    val expected = annotations.map {
+      case _: WriteOutputAnnotationsSpec.Custom => WriteOutputAnnotationsSpec.Replacement(serializedFileName)
+      case a => a
+    }
+
+    val out = phase.transform(annotations)
+
+    info("annotations are unmodified")
+    out.toSeq should be (annotations)
+
+    fileContainsAnnotations(new File(dir, file.toString), expected)
+
+    info(s"file '$serializedFileName' exists")
+    new File(serializedFileName) should (exist)
+  }
+
+  it should "error if multiple annotations try to write to the same file" in new Fixture {
+    val file = new File("write-CustomFileEmission-annotations-error.anno.json")
+    val annotations = Seq( TargetDirAnnotation(dir),
+                           OutputAnnotationFileAnnotation(file.toString),
+                           WriteOutputAnnotationsSpec.Custom("foo"),
+                           WriteOutputAnnotationsSpec.Custom("bar") )
+    intercept[PhaseException] {
+      phase.transform(annotations)
+    }.getMessage should startWith ("Multiple CustomFileEmission annotations")
+  }
+
 }
 
 private object WriteOutputAnnotationsSpec {
+
   case object FooAnnotation extends NoTargetAnnotation
+
   case class BarAnnotation(x: Int) extends NoTargetAnnotation
+
+  case class Custom(value: String) extends NoTargetAnnotation with CustomFileEmission {
+
+    override protected def baseFileName(a: AnnotationSeq): String = "Custom"
+
+    override protected def suffix: Option[String] = Some(".Emission")
+
+    override def toBytes: Option[Stream[Byte]] = Some(new StringBuilder(value).toStream.map(_.toByte))
```

Comment:
```suggestion
    override def toBytes: Option[Iterable[Byte]] = Some(value.toBytes)
```

Reply:
Way simpler, thanks! Fixed in a later commit (and also fixed a few other places where I was doing an overly complex conversion to a string).

(Though, I think the method is `getBytes` not `toBytes`?)

Bikeshedding: Do you think that the `CustomFileEmission` method name should be something different? Apparently `getBytes` is the standard. However, that doesn't return an option. Maybe `getBytesOption` or `getOptionBytes`?

Reply:
It is indeed `getBytes`

Yeah I think maybe changing the name makes sense. I will note that it would be super helpful if we could remember _why_ we made it an `Option`. I guess one argument is that it lets you distinguish between not wanting to write the file and wanting to write an empty file? We should write that in the ScalaDoc.

Reply:
There's no mention of the reason/logic in the dev meeting discussions (2019 or 2020) nor in the commit message. I will note that *I made this an option* so it may have just been developer error on my part.

The only reason I can think of is that this allows use of `replacements` without outputting a file. Note that a `toBytes = None` will result in no file written, currently. A `toBytes = Some(Array.empty)` should be an empty file (I didn't test this).

In the spirit of keeping this moving, how about we switch to `toBytes: Iterable[Byte]`? 

If the hypothetical case of only using `replacements`, this could be an API expansion to `Unserializable`:

```scala
trait Unserializable { this: Annotation => 

  def replacements: AnnotationSeq = Seq.empty

}
```

I think that this also synergizes with your work on serializing annotations with `-ll trace` to JSON to ignore things that don't serialize. This is an approach that could subsume the try/catch logic there.

Reply:
Yeah that sounds neat. If we make `toBytes` an `Iterable[Byte]`, is/should there be a way to make an empty file vs. no file?

Reply:
If you don't want the file, can you just not emit the annotation?

Reply:
Changed to `getBytes: Iterable[Byte]` in added commits.